news-bundle/src/Controller/NewsFeedController.php

```diff
@@ -0,0 +1,257 @@
+        $feed->setLanguage($pageModel->language);
+
+        $time = time();
+        $lastModified = null;
+
+        $archives = StringUtil::deserialize($pageModel->newsArchives);
+        $newsModel = $this->framework->getAdapter(NewsModel::class);
+
+        if ($pageModel->maxFeedItems > 0) {
+            $articles = $newsModel->findPublishedByPids($archives, null, $pageModel->maxFeedItems);
+        } else {
+            $articles = $newsModel->findPublishedByPids($archives);
+        }
+
+        // TODO: Is this still necessary?
+        $origObjPage = $GLOBALS['objPage'] ?? null;
+
+        foreach ($articles as $article) {
+            // Never add unpublished elements to the RSS feeds
+            if (!$article->published || ($article->start && $article->start > $time) || ($article->stop && $article->stop <= $time)) {
+                continue;
+            }
+
+            // TODO: Throw an event here to allow for custom model-to-item transformation?
+            $item = $this->transformModelToItem($article, $pageModel->feedSource, $pageModel);
+
+            if (null === $lastModified) {
+                $lastModified = $item->getLastModified();
+            }
+
+            if ($lastModified < $item->getLastModified()) {
+                $lastModified = $item->getLastModified();
+            }
+
+            $feed->add($item);
+        }
+
+        // TODO: Is this still necessary?
+        $GLOBALS['objPage'] = $origObjPage;
+
+        $feed->setLastModified($lastModified);
+
+        $response = new Response($this->feedIo->format($feed, $pageModel->feedFormat));
+        $response->setLastModified($feed->getLastModified());
+        $response->headers->set('Content-Type', $this->contentTypes[$pageModel->feedFormat]);
+
+        foreach ($archives as $archive) {
+            $this->responseTagger->addTags(['contao.db.news_archive.'.$archive]);
+        }
+
+        return $response;
+    }
+
+    public function configurePageRoute(PageRoute $route): void
+    {
+        $format = $route->getPageModel()->feedFormat;
+        $route->setUrlSuffix($this->urlSuffixes[$format]);
+    }
+
+    public function getUrlSuffixes(): array
+    {
+        return array_unique(array_values($this->urlSuffixes));
+    }
+
+    private function transformModelToItem(NewsModel $model, string $source, PageModel $pageModel): Item
+    {
+        $item = new Item();
+
+        // TODO: Refactor this
+        $link = News::generateNewsUrl($model, false, true);
+
+        $item->setTitle($model->headline)
+            ->setLastModified((new \DateTime())->setTimestamp($model->date))
+            ->setLink($link)
+        ;
+
+        $item->setContent($this->getItemDescription($source, $model, $item));
+
+        /** @var UserModel $authorModel */
+        if (($authorModel = $model->getRelated('author')) instanceof UserModel) {
+            $item->setAuthor((new Author())->setName($authorModel->name));
+        }
```

Comment:
has this already been there? Not sure I like having a backend user name on news items in general, but dunno if a configuration makes sense 🤔

Reply:
No, it has not, but to me this belongs here to be complete. It is also [already done for the schema.org data](https://github.com/contao/contao/blob/5.x/news-bundle/src/Resources/contao/classes/News.php#L392-L399) and it is not the user name but the regular name.

Reply:
true. But for schema.org data (and news templates) I can change it through the template. 🤔